### PR TITLE
Add codecov config

### PR DESCRIPTION
**Pull request by henryre**
See title. Used https://github.com/codecov/example-python as a guide

**Test plan**
Manual verification of Travis and codecov

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -10,19 +10,23 @@ cache: pip
...
+    - name: "Tests: python3.7"
+      python: 3.7
 
 install:
-- pip install -U pip setuptools
-- pip install -U tox==3.12.0
-- tox --notest
+  - pip install -U pip setuptools
+  - pip install -U tox==3.12.0
+  - pip install -U codecov
```
Comment (ajratner) [inline .travis.yml]: Can we run codecov within tox https://github.com/codecov/example-python#testing-with-tox?

File: tox.ini [hunk 2]
```diff
@@ -23,9 +23,14 @@ commands =
 [testenv:type]
 description = run static type checking
 basepython = python3
-usedevelop = True
 commands = mypy -p snorkel
 
+[testenv:coverage]
```
Comment (ajratner) [inline tox.ini]: Note (earlier in file): typo line 12

File: .coveragerc [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+branch = True
+source = snorkel
+
+[report]
+exclude_lines =
+    pragma: no cover
+    raise NotImplementedError
+    if __name__ == .__main__.:
+    def __repr__
+ignore_errors = True
```
Comment (ajratner) [inline .coveragerc]: ?

File: tox.ini (hunk 2 above)
Comment (ajratner) [reply tox.ini]: what kind of coverage tests are these?  Redundant with codecov?

Comment (henryre) [reply tox.ini]: Not a typo: https://tox.readthedocs.io/en/latest/example/basic.html#depending-on-requirements-txt-or-defining-constraints

Comment (henryre) [reply tox.ini]: This command (`pytest --cov=...`) generates the `.coverage` file. The command `codecov` just looks for a file called `.coverage` and uploads it. See https://github.com/codecov/example-python/blob/master/.travis.yml

File: .travis.yml (hunk 1 above)
Comment (henryre) [reply .travis.yml]: Yes, but there's really no reason to. The only place we'd ever run `codecov` is Travis, so I'd prefer to keep it isolated to that environment (as opposed to putting the dep in `requirements-dev.txt` and adding possibly confusing env in `tox.ini`)

File: .coveragerc (hunk 3 above)
Comment (henryre) [reply .coveragerc]: This is the generic .coveragerc suggested by codecov: https://gist.github.com/codecov-io/bf15bde2c7db1a011b6e

File: .travis.yml (hunk 1 above)
Comment (ajratner) [reply .travis.yml]: Okay makes sense

**Pull request merged**